### PR TITLE
Added WatchLater and Home-Page

### PR DESCRIPTION
**Pull request by SaurabhChawda**
Added WatchLater Page and Home-page.
-List of Features
- user can add and remove video from WatchLater.
- Added Dark and Light mode
- Added  Filters in Home-page
Screenshot (364)
Screenshot (363)
Screenshot (362)

### CODE REVIEW

File: src/components/Sidebar/Sidebar.jsx [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
+          <NavLink to="/playlist" className="sidebar__list">
+            <MdPlaylistAdd size={30} color={themeToggle === "light" ? "white" : "black"} />
             <h3 className="sidebar--content">PlayList</h3>
-          </li>
-          <li className="sidebar__list">
-            {/* <img className="sidebar__icon" src="/assets/image/HomePage/Liked.png" alt="Liked" /> */}
-            <AiFillLike size={30} color="white" />
+          </NavLink>
+          <NavLink to="/likedvideos" className="sidebar__list">
+            <AiFillLike size={30} color={themeToggle === "light" ? "white" : "black"} />
```
Comment (harshalini) [inline src/components/Sidebar/Sidebar.jsx:27]: I like that you did it with the help of theme change

File: src/components/Filter/Filter.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        </li>
+        <li className="filter__list-item">
+          <button className="filter__btn" onClick={() => dispatch({ type: "Education", payload: filterableData })}>
+            Education
+          </button>
+        </li>
+        <li className="filter__list-item">
+          <button className="filter__btn" onClick={() => dispatch({ type: "Solutions", payload: filterableData })}>
+            Solutions
+          </button>
```
Comment (harshalini) [inline src/components/Filter/Filter.jsx:37]: If possible, you can create an array of dispatch types and then use map over here for efficiency.

**Review by harshalini — APPROVED** [review]
Code is readable and well structured, I like that color variables are used everywhere. Code looks good to merge

File: src/App.css [hunk 3]
```diff
@@ -16,12 +16,24 @@ a {
...
   --secondary-bg-color: rgb(34, 31, 31);
-  --white-color: white;
-  --black-color: black;
+  --primary-theme-color: white;
+  --secondary-theme-color: black;
+  --font-size-XX-large: clamp(3rem, 12vw, 10rem);
+  --font-size-X-large: clamp(1.2rem, 2.5vw, 10rem);
+  --font-size-large: clamp(1rem, 1.5vw, 10rem);
+  --font-size-regular: clamp(12px, 2.5vw, 1rem);
+  --font-size-small: clamp(8px, 1vw, 1rem);
```
Comment (Bhakti199) [inline src/App.css:29]: liked this. cool

**Review by Bhakti199 — APPROVED** [review]
- This code looks fine to me.
- UI looks cool.
- Liked theme toggle implementation.
- I think remove border shadow. it's distracting. Instead use `1px solid var(--any-color)` primary color(orange). for cards don't use orange border (any other). [This review is according to user expereince.]
- Also I will suggest reduce little bit font-weight.
- maintain border-radius similar enough through-out app.
- give transition 0.3s on hover button.
- Else everything looks fine to me.
- This PR is good to merge.

File: src/components/Sidebar/Sidebar.jsx (hunk 1 above)
Comment (SaurabhChawda) [reply src/components/Sidebar/Sidebar.jsx:27]: ThankYou

**Comment by SaurabhChawda** [comment]
> * This code looks fine to me.
> * UI looks cool.
> * Liked theme toggle implementation.
> * I think remove border shadow. it's distracting. Instead use `1px solid var(--any-color)` primary color(orange). for cards don't use orange border (any other). [This review is according to user expereince.]
> * Also I will suggest reduce little bit font-weight.
> * maintain border-radius similar enough through-out app.
> * give transition 0.3s on hover button.
> * Else everything looks fine to me.
> * This PR is good to merge.

ThankYou so much for deep review, I Will keep in mind and Will make change in Code.

**Pull request merged**